consensus/bor_test/bor_test.go

```diff
@@ -0,0 +1,183 @@
+package bor_test
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"io/ioutil"
+	"math/big"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/maticnetwork/bor/common"
+	"github.com/maticnetwork/bor/consensus/bor"
+	"github.com/maticnetwork/bor/core"
+	"github.com/maticnetwork/bor/core/rawdb"
+	"github.com/maticnetwork/bor/crypto/secp256k1"
+
+	"github.com/maticnetwork/bor/core/types"
+	"github.com/maticnetwork/bor/crypto"
+	"github.com/maticnetwork/bor/eth"
+
+	"github.com/maticnetwork/bor/ethdb"
+	"github.com/maticnetwork/bor/mocks"
+	"github.com/maticnetwork/bor/node"
+)
+
+const (
+	extraSeal = 65 // Fixed number of extra-data suffix bytes reserved for signer seal
+)
+
+type initializeData struct {
+	genesis  *core.Genesis
+	ethereum *eth.Ethereum
+}
+
+func TestCommitSpan(t *testing.T) {
+	init := buildEthereumInstance(t, rawdb.NewMemoryDatabase())
+	chain := init.ethereum.BlockChain()
+	engine := init.ethereum.Engine()
+	_bor := engine.(*bor.Bor)
+
+	// Mock HeimdallClient.FetchWithRetry to return span data from span.json
+	spanData, err := ioutil.ReadFile("span.json")
+	if err != nil {
+		t.Fatalf("%s", err)
+	}
+	res := &bor.ResponseWithHeight{}
+	if err := json.Unmarshal(spanData, res); err != nil {
+		t.Fatalf("%s", err)
+	}
+	h := &mocks.IHeimdallClient{}
+	h.On("FetchWithRetry", "bor", "span", "1").Return(res, nil)
+	_bor.SetHeimdallClient(h)
+
+	// Build 1st blocks header
+	db := init.ethereum.ChainDb()
+	block := init.genesis.ToBlock(db)
+	header := block.Header() // get 0th block header and mutate it as required
+	header.Number = big.NewInt(1)
+	header.ParentHash = block.Hash()
+	header.Time += (init.genesis.Config.Bor.Period + 1)
+	header.Extra = make([]byte, 97) // vanity (32) + extraSeal (65)
+	privKey, _ := hex.DecodeString("b71c71a67e1177ad4e901695e1b4b9ee17ae16c6668d313eac2f96dbcda3f291")
+
+	statedb, err := chain.State()
+	if err != nil {
+		t.Fatalf("%s", err)
+	}
+
+	block, err = _bor.FinalizeAndAssemble(chain, header, statedb, nil, nil, nil)
+	if err != nil {
+		t.Fatalf("%s", err)
+	}
+
+	sig, err := secp256k1.Sign(crypto.Keccak256(bor.BorRLP(header)), privKey)
+	if err != nil {
+		t.Fatalf("%s", err)
+	}
+	copy(header.Extra[len(header.Extra)-extraSeal:], sig)
+
+	block = types.NewBlockWithHeader(header)
+	if _, err := chain.InsertChain([]*types.Block{block}); err != nil {
+		t.Fatalf("%s", err)
+	}
```

Comment:
We should create a utility method that creates new blocks and adds into blockchain for other test cases as well (with header sign etc)

Reply:
Done.